broadinstitute/single_cell_portal_core
# Adding Author and Publication information to Study objects (SCP-2980, SCP-4053)

Currently, the only way for users to add information about authors and publications associated with their study is to do so inside the study description.  While this does make the information searchable, there is no standardized format, and discovering that information is haphazard at best.  This update adds two new models (`Author` and `Publication`) to track that information directly.  In addition, the Study Settings tab has now been refactored to split out into individual tabs for updating various aspects of the study.  Below is a screen shot of the new Settings UX (showing the author/publication tab):

![study_settings_authors_pubs](https://user-images.githubusercontent.com/729968/154513065-ccd67861-3c36-4fb2-9971-74d20b45f963.png)
 
If a study adds any publications/external resources, then a collapsable sidebar will render on the study description tab with links to the relevant papers/resources.  Here is a screenshot of that view (the tab is open by default, but can be closed):

![study_description_pubs_open](https://user-images.githubusercontent.com/729968/154513489-d187fc45-e8a7-4c63-bbed-a8a2cbf7fb10.png)

In addition, this update adds contact links for any corresponding authors to render in the same sidebar.  If an author is marked as corresponding, then a button will render at the top of the sidebar.  Clicking on this button opens a popover with a `mailto:` link inside.  This popover is dynamically rendered into the page, and the email address is protected behind `base64` encoding.  This is to alleviate concerns about screen-scraping email addresses.  Below is a screenshot of what that popover looks like:

![Screen Shot 2022-02-17 at 3 02 37 PM](https://user-images.githubusercontent.com/729968/154562019-6c8f07c2-b8b2-4fd1-a6ac-c33d2af61fd8.png)

MANUAL TESTING
1. Boot as normal and sign in
2. Load any study you have edit access to and open the Study Settings tab
3. Confirm that each sub-tab can be opened, and contains the correct information
4. Go to the authors/publications tab, and enter information about a publication, and add yourself as a corresponding author (you must click `Allow email?` to do so)
5. Click Update settings, and go back to the description view
6. Confirm your paper shows there, there is a button to contact you as an author that opens the popover on click, and that the sidebar can collapse and reopen

This PR continues work outlined in SCP-2980 and satisfies SCP-4053.
